include/boost/math/special_functions/jacobi_theta.hpp

```diff
@@ -0,0 +1,616 @@
+template <class T>
+inline T jacobi_theta2(T z, T q);
+template <class T>
+inline T jacobi_theta3(T z, T q);
+template <class T>
+inline T jacobi_theta4(T z, T q);
+
+// Simple functions - parameterized by tau (assumed imaginary)
+// q = exp(iπτ)
+// tau = -log(q)/π
+template <class T>
+inline T jacobi_theta1tau(T z, T tau);
+template <class T>
+inline T jacobi_theta2tau(T z, T tau);
+template <class T>
+inline T jacobi_theta3tau(T z, T tau);
+template <class T>
+inline T jacobi_theta4tau(T z, T tau);
+
+// Minus one versions for small q / large tau
+template <class T>
+inline T jacobi_theta3m1(T z, T q);
+template <class T>
+inline T jacobi_theta4m1(T z, T q);
+template <class T>
+inline T jacobi_theta3m1tau(T z, T tau);
+template <class T>
+inline T jacobi_theta4m1tau(T z, T tau);
+
+// Policied versions - parameterized by q
+template <class RealType, class Policy>
+inline RealType jacobi_theta1(RealType z, RealType q, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta2(RealType z, RealType q, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta3(RealType z, RealType q, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta4(RealType z, RealType q, const Policy& pol);
+
+// Policied versions - parameterized by tau
+template <class RealType, class Policy>
+inline RealType jacobi_theta1tau(RealType z, RealType tau, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta2tau(RealType z, RealType tau, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta3tau(RealType z, RealType tau, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta4tau(RealType z, RealType tau, const Policy& pol);
+
+// Policied m1 functions
+template <class RealType, class Policy>
+inline RealType jacobi_theta3m1(RealType z, RealType q, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta4m1(RealType z, RealType q, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta3m1tau(RealType z, RealType tau, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta4m1tau(RealType z, RealType tau, const Policy& pol);
+
+// Internal convergence criterion:
+// If the delta is more than an absolute epsilon, keep going.
+// If the delta is more than an relative epsilon, and the denominator
+// is non-zero, keep going.
+template <class RealType>
+inline bool
+_jacobi_theta_converged(RealType result, RealType delta, RealType eps) {
+    return abs(delta) < eps && (abs(result) == 0.0 || abs(delta/result) < eps);
+}
+
+template <class RealType>
+inline RealType
+_jacobi_theta_sum(RealType tau, RealType z_n, RealType z_increment, RealType eps) {
+    RealType delta, partial_result = 0;
+
+    do {
+        delta = exp(-tau*z_n*z_n/constants::pi<RealType>());
```

Comment:
Maybe precompute `tau/constants::pi<RealType>()` so that there isn't a division on each iteration.

People don't realize that elementary special function evaluations are in the same complexity class as division!

Reply:
If we're going to squeeze out divisions like that, it might make sense to divide `tau` by `pi` further up the call stack.